spammer/src/main.rs

```diff
@@ -70,6 +87,63 @@ struct StatsExert {
 const UNIT_KEY: &str = "6c9320ac201caf1f8eaa5b05f5d67a9e77826f3f6be266a0ecccc20416dc6587";
 const DEV_KEY: &str = "1ef7aad365c195462ed04c275d47189d5362bbfe36b5e93ce7ba2f3add5f439b";
 
+#[cfg(feature = "metrics")]
+async fn metrics_handler() -> Result<impl Reply, Rejection> {
+    use prometheus::Encoder;
+    let encoder = prometheus::TextEncoder::new();
+
+    let mut buffer = Vec::new();
+    if let Err(e) = encoder.encode(&REGISTRY.gather(), &mut buffer) {
+        eprintln!("could not encode custom metrics: {}", e);
+    };
+    let mut res = match String::from_utf8(buffer.clone()) {
+        Ok(v) => v,
+        Err(e) => {
+            eprintln!("custom metrics could not be from_utf8'd: {}", e);
+            String::default()
+        }
+    };
+    buffer.clear();
+
+    let mut buffer = Vec::new();
+    if let Err(e) = encoder.encode(&prometheus::gather(), &mut buffer) {
+        eprintln!("could not encode prometheus metrics: {}", e);
+    };
+    let res_custom = match String::from_utf8(buffer.clone()) {
+        Ok(v) => v,
+        Err(e) => {
+            eprintln!("prometheus metrics could not be from_utf8'd: {}", e);
+            String::default()
+        }
+    };
+    buffer.clear();
+
+    res.push_str(&res_custom);
+    Ok(res)
+}
+
+#[cfg(feature = "metrics")]
+async fn register_custom_metrics() {
+    REGISTRY
+        .register(Box::new(TPS.clone()))
+        .expect("collector can be registered");
+
+    REGISTRY
+        .register(Box::new(BLOCK_TIME.clone()))
+        .expect("collector can be registered");
+
+    let metrics_route = warp::path!("metrics").and_then(metrics_handler);
+
+    warp::serve(metrics_route).run(([0, 0, 0, 0], 9501)).await;
+}
+
+#[cfg(feature = "metrics")]
+async fn update_metric_counters(block_time: i64, spammer_tps: i64) {
+    TPS.set(spammer_tps);
+    BLOCK_TIME.set(block_time);
+    BLOCK_NUMBER.inc();
```

Comment:
Can we change this to a gauge, just as block time or block number?
and also obtain the block number from the blockchain itself, because as it is now, if you kill/restart the spammer, the block number will restart at 0 and not from the current block number

Reply:
Done @viquezclaudio